### PR TITLE
Attempt to fix P2 emoji size bug.

### LINKED ISSUE #16868: P2 emoji are extra big
### Screenshot
<kbd></kbd>

### Expected behavior
I would like to see the P2 emoji to be the same size or nearly the same size as emoji that are native to iOS.

### Actual behavior
They look extra big right now, almost 4x larger than they should display.

### Steps to reproduce the behavior
1. Write a P2 comment and include P2 emoji by using :code-test: or :trial: or :eyes: 
2. Visit the comment from Notifications > Comments view.
3. The comment will display with extra big emoji.

Note: This is also reproducible on posts (not just comments)

Previously known and previously fixed issues:
- 1106-gh-Automattic/p2
- https://github.com/wordpress-mobile/WordPress-iOS/issues/15341
- https://github.com/wordpress-mobile/WordPress-iOS/issues/15523
- pbArwn-1v1-p2#comment-2201

##### Tested on iPhone 12 Pro, iOS 14.6, WPiOS 17.7

### PR DESCRIPTION
**Pull request by twstokes**
Fixes #16868

## Description

This is an attempt to work around the large P2 emoji bug by:
- Injecting a CSS style to limit the emoji width to `1 em` if it's on P2
- Detecting P2 emojis when rendering comments and returning their default size

| Before | After |
| - | - |
| <video src="https://github.com/wordpress-mobile/WordPress-iOS/assets/2092798/2ddafb45-e0f5-4e0e-8fc2-5c54808df5a1" /> | <video src="https://github.com/wordpress-mobile/WordPress-iOS/assets/2092798/2bb6d593-c1fd-41ef-86e1-5650b3b43e4d" /> |

**Note:** Gutenberg commenting may be coming soon, which would hopefully overhaul how these are rendered and allow us to remove these workarounds.

## Testing
1. Load this internal P2 in Reader: pdQkYd-1b-p2
2. Expect the output to match the "After" video (emojis are small and similar to web)
3. Tap "View all comments"
4. Expect the output to match the "After" video (emojis are small and similar to web)

- Test on other P2s
- Test loading from Notifications
- Test that the CSS isn't injected on non-P2s (use Safari to observe the web view)

## Regression Notes
1. Potential unintended areas of impact
    - None identified.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
    - Manually tested various P2s.

3. What automated tests I added (or what prevented me from doing so)
    - None.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:
- [x] Portrait and landscape orientations.
- [x] Light and dark modes.
- [x] Fonts: Larger, smaller and bold text. **Note:** Emojis in the post will scale, but emojis in comments will not.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

File: WordPress/Classes/Extensions/URL+Helpers.swift [hunk 1]
```diff
@@ -108,6 +108,10 @@ extension URL {
         return components.count == 4 && isHostedAtWPCom
     }
 
+    var isA8CEmoji: Bool {
```
Comment (aerych) [inline WordPress/Classes/Extensions/URL+Helpers.swift]: Random drive-by nitpicky comment :) Would something like `isWPComEmoji` be more outward facing friendly vs `isA8CEmoji`?

Comment (twstokes) [reply WordPress/Classes/Extensions/URL+Helpers.swift]: Yep! Good call. 👍

Comment (twstokes) [reply WordPress/Classes/Extensions/URL+Helpers.swift]: Completed in d022492b3db3540064d3c84b33831c6f6a025c7c.

**Review by aerych — APPROVED** [review]
Howdy @twstokes 👋 
Makes sense to me, and glad to see this one sorted. I checked with your test p2 and confirmed the emoji are the expected size.  Had some trouble testing other ways. (I'm a little rusty I guess.)  :shipit: from me fwiw.

**Pull request merged**